src/main/java/org/sunbird/budget/service/BudgetServiceImpl.java

```diff
@@ -44,51 +55,52 @@ public SBApiResponse submitBudgetDetails(BudgetInfo data, String userId) throws
 		SBApiResponse response = new SBApiResponse(Constants.API_BUDGET_SCHEME_ADD);
 		try {
 			validateAddBudgetInfo(data);
-			List<BudgetInfoModel> existingList = budgetRepository.getAllByOrgIdAndBudgetYearAndSchemeName(
-					data.getOrgId(), data.getBudgetYear(), data.getSchemeName());
-			if (!CollectionUtils.isEmpty(existingList)) {
+			Map<String, Object> request = new HashMap<>();
+			request.put(Jsonkey.ORG_ID, data.getOrgId());
+			request.put(Jsonkey.BUDGET_YEAR, data.getBudgetYear());
+			request.put(Jsonkey.SCHEME_NAME, data.getSchemeName());
+
+			List<Map<String, Object>> existingDataList = 
+					cassandraOperation.getRecordsByProperties(Jsonkey.DATABASE, Jsonkey.BUDGET_TABLE, request, null);
+
+			if (!existingDataList.isEmpty()) {
 				String errMsg = "Budget Scheme exist for given name. Failed to create BudgetInfo for OrgId: "
 						+ data.getOrgId() + ", BudgetYear: " + data.getBudgetYear() + ", SchemeName: "
 						+ data.getSchemeName();
 				logger.error(errMsg);
 				response.getParams().setErr(errMsg);
 				response.setResponseCode(HttpStatus.BAD_REQUEST);
 				return response;
-			}
-
-			BudgetInfoModel budgetInfoModel = new BudgetInfoModel(
-					new BudgetInfoPrimaryKeyModel(data.getOrgId(), UUID.randomUUID().toString(), data.getBudgetYear()),
-					data.getSchemeName(), data.getSalaryBudgetAllocated(), data.getTrainingBudgetAllocated(),
-					data.getTrainingBudgetUtilization());
-			budgetInfoModel = budgetRepository.save(budgetInfoModel);
-
-			data.setId(budgetInfoModel.getPrimaryKey().getId());
-
-			Audit audit = new Audit(data.getOrgId(), Constants.BUDGET, dateFormatter.format(new Date()), userId,
-					StringUtils.EMPTY, StringUtils.EMPTY, mapper.writeValueAsString(data));
-			audit = auditRepository.save(audit);
-
-			response.getParams().setStatus(Constants.SUCCESSFUL);
-			response.put(Constants.DATA, budgetInfoModel.getBudgetInfo());
-			response.setResponseCode(HttpStatus.CREATED);
+			} 
+
+				request.put(Jsonkey.ID, UUID.randomUUID().toString());
+				request.put(Jsonkey.PROOF_DOCS, data.getProofdocs());
+				request.put(Jsonkey.SALARY_BUDGET_ALLOCATED, data.getSalaryBudgetAllocated());
+				request.put(Jsonkey.TRAINING_BUDGET_ALLOCATED, data.getTrainingBudgetAllocated());
+				request.put(Jsonkey.TRAINING_BUDGET_UTILIZATION, data.getTrainingBudgetUtilization());
+				
+				cassandraOperation.insertRecord(Jsonkey.DATABASE, Jsonkey.BUDGET_TABLE , request);	
+				cassandraOperation.insertRecord(Jsonkey.DATABASE, Jsonkey.AUDIT_TABLE, getAuditMap(userId, request, Constants.CREATE));
+				
+				response.getParams().setStatus(Constants.SUCCESSFUL);
+				response.put(Constants.DATA, request);
```

Comment:
We should read the inserted record again and return it... not the request itself... db record will have proper Id values which is required for UI.